src/contracts/libraries/ShareScalingLib.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "../interfaces/IAVSDirectory.sol";
+
+library ShareScalingLib {
```

Comment:
Why do we need a library for this instead of just calling a function on the AVSDirectory? It complicates the callstack and most of the work is being done in the AVSDirectory anyway..

Is this solely to save on code size?

Reply:
I think it was mainly used as a means to abstracting the scaling logic separately from the contract. These functions are internal so they don't save bytecode size. They are also not pure functions either and depend on reading state from the core contracts.
There's probably not a good enough reason to make this a library as only the DelegationManager will be calling these functions. AVSDirectory simply uses the `INITIAL_TOTAL_MAGNITUDE` constant.

Reply:
AVSDirectory is used however to read the totalMagnitudes